src/LesserPhp/Block.php

```diff
@@ -0,0 +1,124 @@
+<?php
+namespace LesserPhp;
+
+/**
+ * lesserphp
+ * https://www.maswaba.de/lesserphp
+ *
+ * LESS CSS compiler, adapted from http://lesscss.org
+ *
+ * Copyright 2013, Leaf Corcoran <leafot@gmail.com>
+ * Copyright 2016, Marcus Schwarz <github@maswaba.de>
+ * Copyright 2017, Stefan Pöhner <github@poe-php.de>
+ * Licensed under MIT or GPLv3, see LICENSE
+ *
+ * @package LesserPhp
+ */
+
+class Block
+{
+    /**
+     * @var Block|null
+     */
+    public $parent;
```

Comment:
This is replacing the former stdClass object, the properties have to be public, otherwise, the code crashes all over the place.